leshan-core-demo/src/main/java/org/eclipse/leshan/core/demo/cli/converters/ResourceConverter.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Magdalena Kundera
+ *     Orange Polska S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.core.demo.cli.converters;
+
+import org.eclipse.leshan.core.node.InvalidLwM2mPathException;
+import org.eclipse.leshan.core.node.LwM2mNodeUtil;
+import org.eclipse.leshan.core.node.LwM2mPath;
+import org.eclipse.leshan.core.util.Validate;
+
+import picocli.CommandLine.ITypeConverter;
+
+public class ResourceConverter implements ITypeConverter<LwM2mPath> {
```

Comment:
Maybe simple to just extend `LwM2mPathConverter` and just raise an exception if created path is not a path identifying a resource  (see `lwm2mpath.isResource()`?